elastic/kibana
# [Lens] Fix bugs in Lens filters

## Release note

Fixes bugs in Lens display of filters. When navigating to Lens from a dashboard, only pinned filters are used now. Also, when navigating to a previously-saved Lens visualization, Lens now shows all of the filters that were saved with the visualization. Now Lens is accurately showing all saved filters.

## Notes

Fixes https://github.com/elastic/kibana/issues/54941 and fixes https://github.com/elastic/kibana/issues/55603

The bugs are caused by a change to the TopNav component, previously filters were displayed based on Lens's internal state, but now use the `filterManager` component. We did not write code to call `setAppFilters` on the filter manager at the appropriate times.

Steps to verify #55603:

1. Go to a dashboard and add a pinned filter and a regular filter
2. Go to edit mode, and create new > Lens visualization
3. Notice that only the pinned filter is shown in Lens
4. If you go from Lens -> Visualize -> Vertical bar, the pinned filter is the only one shown

Steps to verify #54941:

1. Save a Lens visualization with filters, such as:
<img width="326" alt="Screenshot 2020-01-30 15 31 46" src="https://user-images.githubusercontent.com/666475/73487748-a4013a80-4375-11ea-91f1-4c821183bbdd.png">
2. Refresh the page
